src/main/java/net/starschema/clouddb/jdbc/BQStatement.java

```diff
@@ -190,33 +190,30 @@ private ResultSet executeQueryHelper(String querySql, boolean unlimitedBillingBy
                     (qr.getTotalRows().equals(BigInteger.ZERO) ||
                             (qr.getRows() != null && qr.getTotalRows().equals(BigInteger.valueOf(qr.getRows().size()))));
             // Don't look up the job if we have nothing else we need to do
-            referencedJob = fetchedAll || this.connection.isClosed() ?
-                    null :
-                    qr.getJobReference() == null ?
-                            null :
-                            this.connection.getBigquery()
-                                    .jobs()
-                                    .get(projectId, qr.getJobReference().getJobId())
-                                    .setLocation(qr.getJobReference().getLocation())
-                                    .execute();
+            if (!(fetchedAll || this.connection.isClosed())) {
+                if (qr.getJobReference() != null) {
+                    referencedJob =
+                        this.connection.getBigquery()
+                            .jobs()
+                            .get(projectId, qr.getJobReference().getJobId())
+                            .setLocation(qr.getJobReference().getLocation())
+                            .execute();
+                }
+            }
```

Comment:
I'm a little lost w/o context here - what was causing the NPE?

Reply:
In this particular case, it was https://github.com/jonathanswenson/bqjdbc/blame/0f81d63e747b1e1922bea07bd8936198ecf85ec5/src/main/java/net/starschema/clouddb/jdbc/BQForwardOnlyResultSet.java#L160

But this has been a recurring issue unfortunately :/ It actually turns out that every single getter in the `QueryResponse` object might be null, according to the API, it was just never an issue before because basically the only ones that turn out to be null in practice are `cache_hit` and `total_bytes_processed`.